### DISCUSSION TITLE
Since update on Nextcloud AIO v9.0.1 / Nextcloud 29 - 127.0.0.1 recognized instead of external IP

### DISCUSSION THREAD

**Discussion by neocult-de** [nextcloud/all-in-one#4889] [Questions] [answered]
Does someone else also have the issue, that since the update on Nextcloud AIO v9.0.1 / Nextcloud 29 under Administration -> Security I get the message that I login from 127.0.0.1 instead of external IP on two dedicated Nextcloud-AIO Installations. And since the same update I get the "Local IP" Adress of the Reverse-Proxy in an Reverse-Proxy Setup (3 cases) without changing anything in the configuration myself.

Any ideas, experiences on what have changed?

**Comment by szaimen** [comment, marked as answer]
Hi, can you post the output of `sudo docker exec --user www-data -it nextcloud-aio-nextcloud php occ config:system:get trusted_proxies` here?

**Comment by neocult-de** [comment]
The trusted_proxies returns on all systems (Behind npm reverse-Proxy or dedicated) the following 3 entries which would seem to be default:
127.0.0.1
::1
172.18.0.1/32

**Reply by szaimen** [reply]
Hm... Can you [switch to beta](https://github.com/nextcloud/all-in-one?tab=readme-ov-file#how-to-switch-the-channel) and check if that improves the situation?

**Comment by neocult-de** [comment]
The issue persist in one of the external reverse proxy systems after removing and restarting the beta-mastercontainer with AIO 9.1.0 and stoping and starting and updating the containers (Includes update to Nextcloud 29.0.3).

In the external reverse proxy case:
If I add the reverse proxy IP to trusted_proxies (wasn´t necessary before) then I will get the message that my remote address has been identified as 127.0.0.1

Anything else I can provide or test?

**Comment by szaimen** [comment]
Can you follow https://github.com/nextcloud/all-in-one/discussions/2045?

**Reply by neocult-de** [reply]
I follow that guide on the regular for all my systems wit AIO to activate ipv6 properly. I have the aquivalent of ::1 to 127.0.0.1 if I access my clouds via ipv6.

I redid the steps for ipv6 (including the deleting and reconnecting of the network). I also tried the step to deactivate ipv6 for docker, but it does not change the outcome. (127.0.0.1 recognized)

**Reply by szaimen** [reply]
Have you already tried disabling the userland-proxy once?

**Reply by neocult-de** [reply]
Before I had ipv6 running, redid all the steps.

Then tested ipv4 only as well:

I tried the userland-proxy: false config with just a docker restart and with recreating nextcloud-aio network as follows.
1. Change the /etc/docker/daemon.json  userland-proxy : false  as the only entry
2. sudo service docker restart
3. sudo docker network disconnect nextcloud-aio nextcloud-aio-mastercontainer
4. sudo docker network rm nextcloud-aio
5. sudo docker network create nextcloud-aio
6. sudo docker network connect nextcloud-aio nextcloud-aio-mastercontainer

Anything else I have to keep in mind?

**Comment by neocult-de** [comment]
For testing purposes I would run an Apache docker container with an phpinfo file on the same server as AIO and behind the same reverse-proxy. Given that the apache has the line "RemoteIPHeader X-Forwarded-For" it correctly recognizes my Remote-IP-Adress as would be expected.

Therefore and since I have the same issue in dedicated installations (plural), I would still assume misconfiguration of Caddy/Apache/Nextcloud. 

I might be wrong in the configuration, but it ran in the past, runs with other projects and tests. So any advise welcome. Willing to test more.

**Reply by neocult-de** [reply]
After the "nightly backup and updating the containers" routine, probably the latter step, the issue has been resolved on all instances and remote ip is recognized correctly again. (One of the nights end of June 28-30th).

**Comment by yyasuda59** [comment]
## 原因特定・解決策の共有

PMO ナナイより、ローカルの  リポジトリ（全 1,297 JAR）を全件スキャンした結果を共有します。

---

### 原因

**`secom-commons-1.3.5.jar` / `1.3.6.jar` の内部 POM に `jp.co.secom:parent-pom:1.0.0` への親 POM 参照が埋め込まれています。**

各アプリの `WEB-INF/lib` に配置されているこれらの JAR を `deploy-file` で投入する際、Maven が JAR 内の `META-INF/maven/*/pom.xml` を読み込み、親 POM 解決のために `webgdev9.secom-sis.co.jp` へアクセス → Unknown host で失敗、という流れです。

> ※ `compile_lib` にある `secom-commons-1.3.1.jar` には `<parent>` 参照がなく、`1.3.5` / `1.3.6` から問題が追加されたことが確認できています。

**受領物全体に `parent-pom-1.0.0.pom` の実体は存在しません**（全件スキャンで 0 件）。受領元への追加依頼は不要です。

---

### 解決策（受領元への追加依頼 不要）

`jp.co.secom:parent-pom:1.0.0` のスタブ POM をローカルリポジトリに投入するだけで解決します。Maven は「解決できた」と判断し、`webgdev9` へのアクセスが止まります。



---

### 影響範囲（スタブ投入が必要なアプリ）

以下の 18 アプリは `secom-commons-1.3.5` または `1.3.6` を WEB-INF/lib に持ちます。各アプリの `rewrite-wrapper/repo` に対して上記スタブ投入が必要です。

| アプリ | secom-commons バージョン |
|---|---|
| 1167-IX-coco-dl-adm-main | 1.3.5 |
| 1167-IX-coco-cos-main | 1.3.5 |
| 1167-IX-coco-dl-if-main | 1.3.5 |
| 1167-IX-coco_adm-main | 1.3.5 |
| 1167-IX-msg-api-main | 1.3.5 |
| 1167-IX-pai2-main | 1.3.5 |
| 1167-IX-pai2-fromipgw-main | 1.3.5 |
| 1146-ccmng-01-cccenter-main | 1.3.5 |
| 1146-ccmng-04-LTE-LCA-Int-main | 1.3.5 |
| 1167-IX-cn-rmdl-adm-main | 1.3.5 |
| 1167-IX-coco-mem-release_20260427 | 1.3.6 |
| 1167-IX-mago_adm-main | 1.3.6 |
| 1167-IX-coco_adm3-main | 1.3.6 |
| 1167-IX-kaketuke-kddi-main | 1.3.6 |
| 1167-IX-coco-apply-review_- | 1.3.6 |
| 1167-IX-au-apply-main | 1.3.6 |
| 1167-IX-smp-api-main | 1.3.6 |
| 1167-IX-coco-gm-release_20260423 | ssdt-commons-general-1.1.1 も対象 |

> `secom-commons-1.3.0` / `1.3.1` / `1.3.2` を使うアプリは親 POM 参照なしのため、スタブ投入不要です。

---

### 補足

- **方針 1（失敗アプリをスキップして他を先に回す）は不要になります。** スタブ投入後は全アプリで `dryRun` を試せます。
- スタブ投入はシェルスクリプトで一括化できます。必要であれば作成します。
- 明日確認いただけると助かります 🙏
